numpy/lib/nanfunctions.py

```diff
@@ -607,7 +607,7 @@ def nanargmax(a, axis=None, out=None, *, keepdims=np._NoValue):
 
     """
     a, mask = _replace_nan(a, -np.inf)
-    if mask is not None:
+    if mask is not None and mask.size:
         mask = np.all(mask, axis=axis)
```

Comment:
This seems to occur here, as `np.all` seems to default to `True` when there are no elements along an axis:

```python
In [6]: np.all(np.empty((0, 5)), axis=0)
Out[6]: array([ True,  True,  True,  True,  True])
```